internal/nfs/controller/volume.go

```diff
@@ -273,8 +280,10 @@ func (nv *NFSVolume) getNFSCluster() (string, error) {
 	defer j.Destroy()
 
 	clusterName, err := j.FetchAttribute(nv.ctx, mdPool, nv.objectUUID, clusterNameKey)
-	if err != nil {
-		return "", fmt.Errorf("failed to get cluster name: %w", err)
+	if err != nil && errors.Is(err, util.ErrPoolNotFound) || errors.Is(err, util.ErrKeyNotFound) {
```

Comment:
```suggestion
	if err != nil && (errors.Is(err, util.ErrPoolNotFound) || errors.Is(err, util.ErrKeyNotFound)) {
```

A small nit,
but this should not matter, going by this example
I guess we can strip `err !=nil &&` check completely.
https://go.dev/play/p/e97541eA7HO